chrome/content/zotero/zoteroPane.js

```diff
@@ -4151,74 +4153,277 @@ var ZoteroPane = new function()
 				col,
 				popup,
-				null,
-				(event, collection) => {
-					if (event.target.tagName == 'menuitem') {
-						this.addSelectedItemsToCollection(collection);
-						event.stopPropagation();
-					}
-				},
-				collection => items.every(item => collection.hasItem(item))
+				{
+					clickAction: (event, collection) => {
+						if (event.target.tagName == 'menuitem') {
+							this.addSelectedItemsToCollection(collection);
+							event.stopPropagation();
+						}
+					},
+					disabledCheck: collection => items.every(item => collection.hasItem(item))
+				}
 			);
 			popup.append(menuItem);
 		}
 
-		separator.setAttribute('hidden', !collections.length);
-	};
+		// New Collection
+		addNewCollectionMenuItem(popup, null, Zotero.Libraries.get(Zotero.Libraries.userLibraryID));
+
+		// My Publications
+		let myPublicationMenuItem;
+		if (this.getSelectedLibraryID() == Zotero.Libraries.userLibraryID) {
+			myPublicationMenuItem = document.createXULElement("menuitem");
+			myPublicationMenuItem.setAttribute("label", Zotero.getString("pane.collections.publications"));
+			myPublicationMenuItem.addEventListener("command", (_) => {
+				this.addSelectedItemsToMyPublications();
+			});
+			myPublicationMenuItem.setAttribute("image", "chrome://zotero/skin/16/universal/publications.svg");
+			popup.append(myPublicationMenuItem);
+			// Cannot add top-level notes/attachment or items that are already in my publications
+			if (items.some(item => item.isNote() || item.isAttachment() || item.inPublications)) {
+				myPublicationMenuItem.disabled = true;
+			}
+		}
+
+		let otherLibraries = Zotero.Libraries.getAll().filter(lib => !(lib instanceof Zotero.Feed || lib.libraryID == this.getSelectedLibraryID()));
+		if (!otherLibraries.length) return;
 
+		// Other Libraries
+		let separator = document.createXULElement("menuseparator");
+		let otherLibrariesMenu = document.createXULElement("menu");
+		let otherLibrariesPopup = document.createXULElement("menupopup");
+		otherLibrariesMenu.appendChild(otherLibrariesPopup);
+		document.l10n.setAttributes(otherLibrariesMenu, "item-menu-other-libraries");
+		popup.append(separator, otherLibrariesMenu);
+
+		for (let library of otherLibraries) {
+			let libraryMenu = Zotero.Utilities.Internal.createMenuForTarget(
+				library,
+				otherLibrariesPopup,
+				{
+					clickAction: (event, collection) => {
+						if (event.target.tagName == 'menuitem') {
+							this.addSelectedItemsToCollection(collection);
+							event.stopPropagation();
+						}
+					},
+					disabledCheck: (_) => !library.editable,
+					alwaysMenuForLibrary: true
+				}
+			);
+			otherLibrariesPopup.append(libraryMenu);
+			addNewCollectionMenuItem(libraryMenu, libraryMenu.querySelector("menuseparator"), library);
+		}
+		// Disable destinations where items cannot be added. Essentially a duplicate of
+		// major parts of CollectionTree.canDropCheck and CollectionTree.canDropCheckAsyns
```

Comment:
→ `canDropCheckAsync`

Reply:
For collection copying via context menu, I [had mentioned](https://github.com/zotero/zotero/pull/4420#issuecomment-2406579899) pulling out the `dataType == 'zotero/collection'` stuff to avoid duplicating it and having conditions get out of sync. We didn't do that yet because of time, but ideally we'd do the same thing here for items.

And probably same with the stuff from `onDrop` below. We really don't want to have copying code in multiple places if we can help it, among other things because this stuff will definitely be changing as we improve cross-library copying, and we always want drag-and-drop and the menus to trigger the same UI and logic. This, not localization, is the place for DRY!

Reply:
Right, totally! I believe, while putting this together, the a11y review was still happening and I wasn't sure if these changes would or would not be required to finish it up. So I did go the same route as we did in #4420, thinking of it as a quick followup to other drag-drop alternatives PRs. But now that VPAT is settled, there's no reason to not refactor things to avoid duplication, for sure. I'll go through this logic to move it to `Zotero.Collection` and have it be used here and in collectionTree during drag-drop.

Reply:
So maybe something like this? I pulled out the `canDropCheck` and `canDropCheckAsync` into `Zotero.Item.canAddToContainer` and `Zotero.Item.canAddToContainerAsync` that both `collectionTree.onDrop` and contextmenus from `ZoteroPane` share. I think in terms of readability and overall structure, it is a big improvement!

While we are at it, would it make sense to do the same for collections drag-drop as well here? Or is it better to do in a separate PR, since it's technically not relevant to these changes.